src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_IndexerAccess.cs

```diff
@@ -220,37 +220,16 @@ private BoundSequence VisitIndexOrRangePatternIndexerAccess(BoundIndexOrRangePat
                 out var receiverStore,
                 // Store the receiver as a ref local if it's a value type to ensure side effects are propagated
                 receiver.Type.IsReferenceType ? RefKind.None : RefKind.Ref);
-            var lengthLocal = F.StoreToTemp(F.Property(receiverLocal, lengthOrCountProperty), out var lengthStore);
-            var indexLocal = F.StoreToTemp(
-                MakePatternIndexOffsetExpression(argument, lengthLocal, out bool usedLength),
-                out var indexStore);
-
-            // Hint the array size here because the only case when the length is not needed is if the
-            // user writes code like receiver[(Index)offset], as opposed to just receiver[offset]
-            // and that will probably be very rare.
-            var locals = ArrayBuilder<LocalSymbol>.GetInstance(3);
-            var sideEffects = ArrayBuilder<BoundExpression>.GetInstance(3);
-
-            locals.Add(receiverLocal.LocalSymbol);
-            sideEffects.Add(receiverStore);
-
-            if (usedLength)
-            {
-                locals.Add(lengthLocal.LocalSymbol);
-                sideEffects.Add(lengthStore);
-            }
-
-            locals.Add(indexLocal.LocalSymbol);
-            sideEffects.Add(indexStore);
+            var indexAccess = MakePatternIndexOffsetExpression(argument, F.Property(receiverLocal, lengthOrCountProperty), out _);
```

Comment:
Should the "Lowered code:" comment above be adjusted?

Reply:
> Should the "Lowered code:" comment above be adjusted?

I'll remove the comment because it is redundand.